### PR TITLE
fix(codecs): fix 'ProtobufSerializerConfig' input type

### PR DESCRIPTION
**Pull request by pront**
closes: https://github.com/vectordotdev/vector/issues/19230

### CODE REVIEW

**Review by bruceg — COMMENTED** [review]
This slipped through testing once already, could we set up a unit test to ensure it stays fixed?

File: lib/codecs/src/encoding/format/protobuf.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl ProtobufSerializerConfig {
 
     /// The data type of events that are accepted by `ProtobufSerializer`.
     pub fn input_type(&self) -> DataType {
-        DataType::Log.and(DataType::Trace)
+        DataType::Log.and(DataType::all())
```
[the remark this answers is not in the document]
Comment (pront) [reply lib/codecs/src/encoding/format/protobuf.rs]: Good point. This should simply be `DataType::all()`, similar to native_json.

**Comment by pront** [comment]
> This slipped through testing once already, could we set up a unit test to ensure it stays fixed?

Added a simple test https://github.com/vectordotdev/vector/pull/19264/commits/b9922e3ff4130f9e0001dc95fcc8a02f98a861e5. This should bring awareness next time someone changes the input type. If you have other ideas, please let me know and I am happy to follow up.

File: lib/codecs/src/encoding/format/protobuf.rs [hunk 2]
```diff
@@ -31,7 +31,7 @@ impl ProtobufSerializerConfig {
 
     /// The data type of events that are accepted by `ProtobufSerializer`.
     pub fn input_type(&self) -> DataType {
-        DataType::Log.and(DataType::Trace)
+        DataType::all()
```
Comment (jszwedko) [inline lib/codecs/src/encoding/format/protobuf.rs:34]: Can this codec encode metrics? That would surprise me a little. I think this codec may only be applicable to logs.

Comment (jszwedko) [reply lib/codecs/src/encoding/format/protobuf.rs:34]: Consider that the `protobuf` decoder only emits logs: https://github.com/vectordotdev/vector/blob/92d4102df59f3239b9aa062caa28e19ead7f3c57/lib/codecs/src/decoding/format/protobuf.rs#L41-L43

File: lib/codecs/src/encoding/format/protobuf.rs [hunk 3]
```diff
@@ -234,12 +234,26 @@ mod tests {
...
 
+    #[test]
+    fn test_config_input_type() {
+        let config = ProtobufSerializerConfig {
+            protobuf: ProtobufSerializerOptions {
+                desc_file: test_data_dir().join("test_protobuf.desc"),
+                message_type: "test_protobuf.Person".into(),
+            },
+        };
+        assert_eq!(config.input_type(), DataType::all());
```
Comment (bruceg) [inline lib/codecs/src/encoding/format/protobuf.rs:254]: With `fn input_type` returning `Log`, I think this should fail.

**Comment by pront** [comment]
> Does this actually fix the issue identified in #19230 then? Or do we need a follow-up?

It does fix the issue. I added a repro case [here](https://github.com/vectordotdev/issue-examples/tree/main/issues/19230) which fails on master but works with this branch. The problem was that all events were [filtered out](https://github.com/vectordotdev/vector/blob/master/src/topology/builder.rs#L603C5-L603C5) before they even reach the console sink. Now with this fix, the events reach the sink and the protobuf encoding itself works.

**Comment by jszwedko** [comment]
> > Does this actually fix the issue identified in #19230 then? Or do we need a follow-up?
> 
> It does fix the issue. I added a repro case [here](https://github.com/vectordotdev/issue-examples/tree/main/issues/19230) which fails on master but works with this branch. The problem was that all events were [filtered out](https://github.com/vectordotdev/vector/blob/master/src/topology/builder.rs#L603C5-L603C5) before they even reach the console sink. Now with this fix, the events reach the sink and the protobuf encoding itself works.

Aaah, I see it now. It was using `&` rather than `|` to join `Log` and `Trace`.

**Pull request merged**